File: src/types/open_order.ts
export interface OpenOrder {
  id: string;
  table_id: string | null;
  session_id: string | null;
  note: string | null;
  created_at: string;
  updated_at: string;
}

export interface OpenOrderLine {
  id: string;
  order_id: string;
  line_no: number;
  product_id: string | null;
  product_name: string;
  product_sku: string | null;
  quantity: number;
  unit_price_ttc: number;
  unit_price_ht: number;
  tva_rate_pct: number;
  discount_ttc: number;
}

export interface OpenOrderFull {
  order: OpenOrder;
  lines: OpenOrderLine[];
}

export interface OpenOrderLineInput {
  product_id: string | null;
  product_name: string;
  product_sku: string | null;
  quantity: number;
  unit_price_ttc: number;
  unit_price_ht: number;
  tva_rate_pct: number;
  discount_ttc: number;
}

File: src-tauri/src/db/models/open_order.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct OpenOrder {
    pub id: String,
    pub table_id: Option<String>,
    pub session_id: Option<String>,
    pub note: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct OpenOrderLine {
    pub id: String,
    pub order_id: String,
    pub line_no: i64,
    pub product_id: Option<String>,
    pub product_name: String,
    pub product_sku: Option<String>,
    pub quantity: i64,
    pub unit_price_ttc: i64,
    pub unit_price_ht: i64,
    pub tva_rate_pct: i64,
    pub discount_ttc: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OpenOrderFull {
    pub order: OpenOrder,
    pub lines: Vec<OpenOrderLine>,
}

/// Entrée envoyée par le frontend pour sauvegarder une ligne.
#[derive(Debug, Deserialize)]
pub struct OpenOrderLineInput {
    pub product_id: Option<String>,
    pub product_name: String,
    pub product_sku: Option<String>,
    pub quantity: i64,
    pub unit_price_ttc: i64,
    pub unit_price_ht: i64,
    pub tva_rate_pct: i64,
    pub discount_ttc: i64,
}
